src/GovUk.Education.ExploreEducationStatistics.Data.Model/Migrations/20200930085741_Routine_GetFilteredObservations.sql

```diff
@@ -0,0 +1,164 @@
+CREATE OR ALTER PROCEDURE GetFilteredObservations
+    @geographicLevel nvarchar(6) = NULL,
+    @timePeriodList TimePeriodListType READONLY,
+    @countriesList IdListVarcharType READONLY,
+    @institutionsList IdListVarcharType READONLY,
+    @localAuthorityList IdListVarcharType READONLY,
+    @localAuthorityOldCodeList IdListVarcharType READONLY,
+    @localAuthorityDistrictList IdListVarcharType READONLY,
+    @localEnterprisePartnershipsList IdListVarcharType READONLY,
+    @mayoralCombinedAuthoritiesList IdListVarcharType READONLY,
+    @multiAcademyTrustList IdListVarcharType READONLY,
+    @opportunityAreasList IdListVarcharType READONLY,
+    @parliamentaryConstituenciesList IdListVarcharType READONLY,
+    @regionsList IdListVarcharType READONLY,
+    @rscRegionsList IdListVarcharType READONLY,
+    @sponsorList IdListVarcharType READONLY,
+    @wardsList IdListVarcharType READONLY,
+    @planningAreasList IdListVarcharType READONLY,
+    @filterItemList IdListGuidType READONLY
+AS
+
+DECLARE @filterListTemp AS TABLE (RowID INT NOT NULL PRIMARY KEY IDENTITY(1,1), FilterId uniqueidentifier);
+INSERT INTO @filterListTemp SELECT DISTINCT FilterId FROM ObservationFilterItem WHERE FilterItemId in (select id from @filterItemList);
```

Comment:
It might be a lot quicker to execute this to build the filter temp list rather than select from `ObservationFilterItem `
```
SELECT DISTINCT F.Id
FROM Filter F
JOIN FilterGroup FG ON F.Id = FG.FilterId
JOIN FilterItem FI ON FG.Id = FI.FilterGroupId
WHERE FI.Id IN (SELECT ID FROM @filterItemList)
```

Reply:
Yep - will try that as it was very slow!

Reply:
done